homeassistant/components/overkiz/config_flow.py

```diff
@@ -351,8 +351,9 @@ async def async_step_reauth(
             self.hass.config_entries.async_get_entry(self.context["entry_id"]),
         )
 
+        # overkiz entries always have unique IDs
         self.context["title_placeholders"] = {
-            "gateway_id": self._reauth_entry.unique_id
+            "gateway_id": cast(str, self._reauth_entry.unique_id)
```

Comment:
Can you elaborate why this change is required? Is `unique_id not always a string within HA? The cast will work indeed, because we always set the unique id to the gateway_id which is a string.

Happy to approve, but would love to learn why this change was required.

Reply:
unique id is `str | None` (we don't require config entries to have a unique id)

This PR is in preparation for PR https://github.com/home-assistant/core/pull/126291 which makes the flow context a typed dictionary.

In case of overkiz we know the unique_id is not `None`, hence the cast